codex-rs/login/src/lib.rs

```diff
@@ -63,8 +63,11 @@ impl CodexAuth {
 
     /// Loads the available auth information from the auth.json or
     /// OPENAI_API_KEY environment variable.
-    pub fn from_codex_home(codex_home: &Path) -> std::io::Result<Option<CodexAuth>> {
-        load_auth(codex_home, true)
+    pub fn from_codex_home(
+        codex_home: &Path,
+        always_use_api_key_signing: bool,
```

Comment:
Seeing this passed in as a raw `bool` value is not great. Please introduce an enum with two variants like:

```rust
enum AuthCredentialStrategy {
    ApiKeyOnly,
    PreferChatGpt,
}
```

or whatever names you think are appropriate. Then update this function to take the new enum type instead of a `bool`.